test-notebooks/Dask_Cluster_Benchmark_notebook.ipynb

```diff
@@ -0,0 +1,166 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "935e9a49-882e-4d9f-bc8f-a02aa2373171",
+   "metadata": {},
+   "source": [
+    "<h1>Distributed vs Non-Distributed Benchmark</h1>\n",
+    "\n",
+    "This test will be used to benchmark a direct comparison of processing time for a task between a singleuser instance and a dask-work cluster."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "56477b25-751f-4eab-8514-52d604f336cf",
+   "metadata": {},
+   "source": [
+    "<h2>Dask Gateway</h2>\n",
+    "Dask Gateway provides a secure, multi-tenant server for managing Dask clusters. It allows users to launch and use Dask clusters in a shared, centrally managed cluster environment, without requiring users to have direct access to the underlying cluster backend (e.g. Kubernetes, Hadoop/YARN, HPC Job queues, etc…).\n",
+    "\n",
+    "Dask Gateway is one of many options for deploying Dask clusters, see Deploying Dask in the Dask documentation for an overview of additional options.\n",
+    "\n",
+    "<h3>Highlights</h3>\n",
+    "\n",
+    "* Centrally Managed: Administrators do the heavy lifting of configuring the Gateway, users simply connect to the Gateway to get a new cluster. Eases deployment, and allows enforcing consistent configuration across all users.\n",
+    "\n",
+    "* Secure by Default: Cluster communication is automatically encrypted with TLS. All operations are authenticated with a configurable protocol, allowing you to use what makes sense for your organization.\n",
+    "\n",
+    "* Flexible: The gateway is designed to support multiple backends, and runs equally well in the cloud as on-premise. Natively supports Kubernetes, Hadoop/YARN, and HPC Job Queueing systems.\n",
+    "\n",
+    "* Robust to Failure: The gateway can be restarted or experience failover without losing existing clusters. Allows for seamless upgrades and restarts without disrupting users."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "13f24e39-bc62-4539-a15b-81c1639209e0",
+   "metadata": {},
+   "source": [
+    "<h3>Architecture Overview</h3>\n",
+    "Dask Gateway is divided into three separate components:\n",
+    "\n",
+    "Multiple active Dask Clusters (potentially more than one per user)\n",
+    "\n",
+    "A Proxy for proxying both the connection between the user’s client and their respective scheduler, and the Dask Web UI for each cluster\n",
+    "\n",
+    "A central Gateway that manages authentication and cluster startup/shutdown\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "430bc176-6595-4942-b78b-37922920abb9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import dask, time, torch\n",
+    "from dask_gateway import Gateway\n",
+    "import time\n",
+    "import torch\n",
+    "gateway = Gateway( \"http://10.107.108.18\", auth=\"jupyterhub\", )\n",
+    "options = gateway.cluster_options()"
```

Comment:
Not entirely sure if this options section is necessary because as of now none of the options are actually being exposed to the user to modify. May not be important but I never use it when testing my code.